File: 0882-peak-index-in-a-mountain-array/solution.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int start = 0;
        int end = arr.length - 1;
        while(end >= start){
            int mid = (start + end) / 2;
            if(mid+1 < arr.length && arr[mid+1] > arr[mid]){
                start = mid + 1;
            }
            else if(mid-1 > 0 && arr[mid-1] > arr[mid]){
                end = mid - 1;
            }
            else{
                return mid;
            }
        }
        return 1;
    }
}

File: 0882-peak-index-in-a-mountain-array/solution.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
        start=0
        end = len(arr)-1
        while(start<=end):
            mid = (start+end)//2
            if(arr[mid-1]<arr[mid] and arr[mid]>arr[mid+1]):
                return mid
            elif(arr[mid-1]>arr[mid]):
                end=mid
            else:
                start=mid+1
        return start
        
